### PR TITLE
`tensorflow` add `tensorflow.saved_model`

### PR DESCRIPTION
**Pull request by hoel-bagard**
Some of the stubs are taken from [here](https://github.com/hmc-cs-mdrissi/tensorflow_stubs/blob/main/stubs/tensorflow/nn.pyi).

All the members from the `tensorflow.python` module are causing the CI to fail since
```python
from tensorflow import python
python.trackable.resource.CapturableResource  # Ok

import tensorflow
tensorflow.python.trackable.resource.CapturableResource  # AttributeError: module 'tensorflow' has no attribute 'python'
````

Should the whole `tensorflow.python` module be added to the allowlist ? I don't understand why the CI is fine with the other `tensorflow.python` modules that are already in the stubs.

### CODE REVIEW

**Comment by hoel-bagard** [comment]
@hmc-cs-mdrissi If you have time to have review it.

File: stubs/tensorflow/tensorflow/python/trackable/ressource.pyi [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from tensorflow.python.trackable.base import Trackable
+
+class _ResourceMetaclass(type): ...
+
+# Internal type that is commonly used as a base class
+# and some public apis the signature needs it.
+class CapturableResource(Trackable, metaclass=_ResourceMetaclass): ...
```
Comment (srittau) [inline stubs/tensorflow/tensorflow/python/trackable/ressource.pyi:7]: Not a full review:

There is no file `ressource.py` in the source. Did you mean to name this file `resource.pyi`? Also, `resource.py` contains more items that the stubs. If you don't want to add them add this time, please add the following to the stubs:

```python
from _typeshed import Incomplete

def __getitem__(name: str) -> Incomplete: ...  # incomplete
```

Comment (hoel-bagard) [reply stubs/tensorflow/tensorflow/python/trackable/ressource.pyi]: Fixed in bd198e492, thanks!

File: stubs/tensorflow/tensorflow/python/trackable/ressource.pyi [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from tensorflow.python.trackable.base import Trackable
+
+class _ResourceMetaclass(type): ...
+
+# Internal type that is commonly used as a base class
+# and some public apis the signature needs it.
```
[the remark this answers is not in the document]
Comment (JelleZijlstra) [reply stubs/tensorflow/tensorflow/python/trackable/ressource.pyi]: This was an old review I forgot to submit, will reapply it in the right place.

File: stubs/tensorflow/tensorflow/python/trackable/resource.pyi [hunk 3]
```diff
@@ -0,0 +1,11 @@
+from _typeshed import Incomplete
+
+from tensorflow.python.trackable.base import Trackable
+
+class _ResourceMetaclass(type): ...
+
+# Internal type that is commonly used as a base class
+# and some public apis the signature needs it.
```
Comment (JelleZijlstra) [inline stubs/tensorflow/tensorflow/python/trackable/resource.pyi:8]:
```suggestion
# it is needed for the public signatures of some APIs.
```

File: stubs/tensorflow/tensorflow/saved_model/__init__.pyi [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    experimental_io_device: str
+    experimental_variable_policy: VariablePolicy
+    experimental_custom_gradients: bool
+    experimental_image_format: bool
+    experimental_skip_saver: bool
+    def __init__(
+        self,
+        namespace_whitelist: list[str] | None = None,
+        save_debug_info: bool = False,
+        function_aliases: dict[str, tf.types.experimental.GenericFunction[..., object]] | None = None,
```
Comment (hmc-cs-mdrissi) [inline stubs/tensorflow/tensorflow/saved_model/__init__.pyi:60]: Minor nit: Can this be Mapping instead?

**Review by hmc-cs-mdrissi — APPROVED** [review]
Overall looks quite good.

File: stubs/tensorflow/tensorflow/saved_model/__init__.pyi (hunk 4 above)
Comment (hoel-bagard) [reply stubs/tensorflow/tensorflow/saved_model/__init__.pyi:60]: Done in 06b74530b.

**Pull request merged**